src/EditorFeatures/Core/EditorConfigSettings/Data/AnalyzerSetting.cs

```diff
@@ -60,5 +66,20 @@ internal void ChangeSeverity(DiagnosticSeverity severity)
             Severity = severity;
             _settingsUpdater.QueueUpdate(this, severity);
         }
+
+        public string GetSettingName()
+        {
+            return $"dotnet_diagnostic.{Id}.severity";
+        }
+
+        public string GetDocumentation()
+        {
+            return Description;
+        }
+
+        public string[] GetSettingValues(OptionSet optionSet)
```

Comment:
passing out arrays is not ideal.  could this be an `ImmutableArray<string>` instead?